src/frontends/pytorch/src/input_model.cpp

```diff
@@ -0,0 +1,127 @@
+        auto type_any = simplified_type_interpret(m_model_decoder->get_input_type(i));
+        auto dtype = element::dynamic;
+        if (type_any.is<element::Type>()) {
+            dtype = type_any.as<element::Type>();
+        }
+        m_descriptors.emplace(inputs[i], PlaceDesc(dtype, m_model_decoder->get_input_shape(i)));
+    }
+    const auto& outputs = m_model_decoder->outputs();
+    for (size_t i = 0; i < outputs.size(); ++i) {
+        auto out_place = std::make_shared<pytorch::Place>(*this, outputs[i]);
+        for (const auto& name : out_place->get_names()) {
+            m_name_to_place.emplace(name, std::dynamic_pointer_cast<frontend::Place>(out_place));
+        }
+        auto type_any = simplified_type_interpret(m_model_decoder->get_output_type(i));
+        auto dtype = element::dynamic;
+        if (type_any.is<element::Type>()) {
+            dtype = type_any.as<element::Type>();
+        }
+        m_descriptors.emplace(outputs[i], PlaceDesc(dtype, m_model_decoder->get_output_shape(i)));
+    }
+}
+
+std::vector<ov::frontend::Place::Ptr> InputModel::get_inputs() const {
+    std::vector<ov::frontend::Place::Ptr> res;
+    for (const auto& input_idx : m_model_decoder->inputs()) {
+        auto place_it = m_name_to_place.find(std::to_string(input_idx));
+        FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for input.");
+        res.push_back(place_it->second);
+    }
+    return res;
+}
+
+std::vector<ov::frontend::Place::Ptr> InputModel::get_outputs() const {
+    std::vector<ov::frontend::Place::Ptr> res;
+    for (const auto& output_idx : m_model_decoder->outputs()) {
+        auto place_it = m_name_to_place.find(std::to_string(output_idx));
+        FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for output.");
+        res.push_back(place_it->second);
+    }
+    return res;
+}
+
+Place::Ptr InputModel::get_place_by_tensor_name(const std::string& tensor_name) const {
+    auto place_it = m_name_to_place.find(tensor_name);
+    FRONT_END_GENERAL_CHECK(place_it != m_name_to_place.end(), "Couldn't find Place for provided tensor name.");
+    return place_it->second;
+}
+
+void InputModel::set_partial_shape(const Place::Ptr& place, const ov::PartialShape& shape) {
+    FRONT_END_GENERAL_CHECK(place->is_input(), "Setting partial shape is supported for input only.");
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
+    auto it = m_descriptors.find(pytorch_place->get_tensor_index());
+    if (it != m_descriptors.end()) {
+        it->second.m_pshape = shape;
+    }
+}
+
+ov::PartialShape InputModel::get_partial_shape(const Place::Ptr& place) const {
+    auto pytorch_place = std::dynamic_pointer_cast<pytorch::Place>(place);
+    FRONT_END_GENERAL_CHECK(pytorch_place, "Only place produced by PyTorch Frontend is supported");
```

Comment:
I think more user-friendly message should be that you support setting shapes to only model inputs. This messages does not tell anything to the user:)
The same comment for other methods

Reply:
Agree that for the case of `nullptr` it is misleading